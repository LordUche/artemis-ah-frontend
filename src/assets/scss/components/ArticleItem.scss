$border-radius: 5px;

.article-item {
    box-shadow: 2px 2px 10px rgba(0,0,0,0.25);
    margin-top: 20px;
    position: relative;
    border-radius: $border-radius;

    .stats-wrapper {
        position: absolute;
        background: #333333;
        color: #ffffff;
        padding: 5px 7px;
        font-size: 13px;
        border-top-right-radius: $border-radius;

        &__read-time {
            margin-left: 10px;
        }
    }

    & > .stats-wrapper {
        display: none;
    }

    &__image-wrapper {
        position: relative;

        .tag {
            position: absolute;
            top: 10px;
            left: -10px;
            background: #ffffff;
            padding: 5px 10px;
            box-shadow: 0 2px 5px rgba(0, 0, 0, 0.35);
            font-size: 14px;
            cursor: pointer;
        }

        img {
            width: 100%;
            border-radius: $border-radius;
            border-bottom-left-radius: 0;
            border-bottom-right-radius: 0;
            box-shadow: 0px 2px 10px rgba(0, 0, 0, 0.25)
        }

        .stats-wrapper {
            bottom: 4px;
            left: 0;
        }
    }

    &__body-wrapper {
        padding-bottom: 40px;

        &__body {
            padding: 10px;
        
            &__title {
                font-weight: bold;
                font-size: 17px;
                line-height: 100%;
    
                a {
                    color: #333333;
                    display: block;
                    width: calc(100% - 140px);
    
                    &:hover {
                        color: #000000;
                    }
                }
            }

            &__author {
                margin-top: 5px;
                font-size: 14px;

                a {
                    font-size: inherit;
                    line-height: 100%;
                    color: #787878;
                }
            }
    
            &__description {
                color: #444444;
                margin-top: 8px;
                font-size: 14px;
            }
        }
    
        &__bottom-links {
            position: absolute;
            left: 10px;
            right: 10px;
            bottom: 10px;
            display: flex;
            justify-content: space-between;
            align-items: flex-end;
    
            &__user-actions {
                float: right;
                margin-bottom: -10px;

                &__edit {
                    font-size: 1.5rem;
                    color: #000000;
                    box-shadow: #ffffff;

                    transition-property: box-shadow;
                    transition-duration: 0.2s;
                    transition-timing-function: ease-in-out;

                    transition-property: font-size;
                    transition-duration: 0.1s;
                    transition-timing-function: ease-in-out;
                }
                &__edit:hover, &__delete:hover {
                    font-size: 1.41rem;
                    box-shadow: 1px 1px 1px 1px #787878;
                    cursor: pointer;
                }

                &__delete {
                    font-size: 1.5rem;
                    color: #F94A4A;
                    margin-left: 0.5rem;
                    box-shadow: #ffffff;

                    transition-property: box-shadow;
                    transition-duration: 0.2s;
                    transition-timing-function: ease-in-out;

                    transition-property: font-size;
                    transition-duration: 0.1s;
                    transition-timing-function: ease-in-out;
                }
            }
        }
    }

    @media screen and (min-width: 769px) {
        margin-top: 50px;
        height: 159px;

        & > .stats-wrapper {
            top: 0;
            right: 0;
            display: block;
        }

        &__image-wrapper {
            width: 180px;
            background: #f9f9f9;
            height: 189px;
            margin-top: -30px;
            float: left;
    
            img {
                width: 100%;
                height: 100%;
                vertical-align: top;
                border-radius: $border-radius;
                border-bottom-right-radius: 0;
            }

            .stats-wrapper {
                bottom: 0;
                left: 0;
                display: none;
            }
        }

        &__body-wrapper {
            height: 159px;
            margin-left: 180px;
            padding-bottom: 0;
            position: relative;
    
            &__body {
                padding: 10px 0 0 10px;

                &__title,
                &__description {
                    overflow: hidden;
                    display: -webkit-box;
                    -webkit-line-clamp: 2;
                    -webkit-box-orient: vertical;
                    text-overflow: ellipsis;
                }

                &__title {
                    max-height: 38px;
                }

                &__description {
                    max-height: 35px;
                }
            }
    
            &__bottom-links {
                position: absolute;
                left: 10px;
                right: 10px;
                bottom: 10px;
    
                &__read-more {
                    font-size: 13px;
                }
            }
        }

        &:after {
            content: '';
            display: block;
            clear: both;
        }
    
        .stats-wrapper {
            position: absolute;
            top: 0;
            right: 0;
            background: #333333;
            color: #ffffff;
            padding: 5px 7px;
            font-size: 13px;
            border-top-right-radius: $border-radius;
    
            &__read-time {
                margin-left: 10px;
            }
        }
    }
}

.default-toast {
    font-size: 1.5rem;
    text-transform: capitalize;
    color: $primary-color;
    background: white;
}
