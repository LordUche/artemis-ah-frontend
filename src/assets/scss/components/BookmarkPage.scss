.bookmark_section{
  padding: 20px 5% 30px;
  width: 100%;
  height: 60vh;
  overflow: auto;

  &_empty_text{
    margin: 10% auto;
    font-size: 16px;
    opacity: .5;
    width: 271px;
  }

  .article-item{
    &:last-of-type{
    margin-bottom: 100px;
    }
  }

  .pagination{
    margin-top: 30px;
  }
}

