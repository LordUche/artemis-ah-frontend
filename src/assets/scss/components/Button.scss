.ah-btn , a.ah-btn {
  border-radius: 5px;
  padding: 10px 15px;
  cursor: pointer;
  font-size: 15px;
  font-family: "Nunito", sans-serif;

  img {
   width: 15px;
   height: 15px;
   margin-right: 10px;
  }

  &[disabled] {
   opacity: 0.5;
   cursor: not-allowed;
  }
 }
 .ah-btn:hover {
  opacity: 0.9;;
 }

a.ah-btn {
  display: inline-block;
}
