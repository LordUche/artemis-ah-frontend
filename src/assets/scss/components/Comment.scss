@import "./base.scss";

.comment {
  width: 60%;
  margin: auto;
  font-size: 1.6rem;

  .like-icon{
    cursor: pointer;
  }

  .liked{
    color: #06A5E9;
  }

  &_box{
    width: 100%;
    margin-bottom: 5%;

    &_post {
      padding: 10px 0;
      float: right;
      color: #2fc1ff;
      outline-color: #cccccc;
      font-weight: $bold-font;
    }

    &_post:hover {
      cursor: pointer;
    }

    &__form {
      width: 100%;

      &__input {
        width: 100%;
        height: 150px;
        padding: 10px;
        margin-bottom: 1%;
        border-radius: 5px;
        overflow: hidden;
        overflow-y: scroll;
      }

      &__submit {
        float: right;
        box-shadow: 0 0 2px #000000;
        background: $primary-color;
        color: $white-color;
        width: 20%;
      }

      &__cancel {
        float: right;
        background: #ffffff;
        box-shadow: 0 0 2px #000000;
        width: 20%;
        margin-left: 10px;
      }
    }
  }
}
.comment_card {
  width: 100%;
  padding: 1% 0;
  margin: 2% 0;
  display: grid;
  grid-template-columns: 10% 90%;
  border-bottom: 1px solid #f4f4f4;

  &__image {
    text-align: center;
    img {
      width: 50px;
      height: 50px;
      object-fit: cover;
      border-radius: 100%;
    }
  }

  &__main {
    &__header {
      h3 {
        color: #2fc1ff;
        text-transform: capitalize;
      }

      i {
        font-size: 12px;
        color: #888888;
      }
    }

    &__body {
      margin: 2% 0;
    }

    &__footer {
      color: #888888;
      position: relative;
    }

    &__likes {
      padding-left: 5px;
      font-family: "Nunito", "Courier New", Courier, monospace;
    }
  }
}

.hidePost {
  display: none;
}
.no_comment {
  margin: 3% 0;
  font-style: italic;
  color: #999999;
}

.full_comment_wrapper {
  margin-bottom: 5em;
}
.edit_comment_container {
  width: 100%;
}

.comment_input_form {
  width: 100%;
}

.comment_input_form #edit_comment_textarea {
  width: 100%;
}

.edit_comment_button_wrapper {
  text-align: right;
}

.view_comment_history_toggle {
  cursor: pointer;
  font-weight: bold;
  position: absolute;
  right: 6em;
}
.view_comment_history_toggle:hover {
  text-shadow: 0.5px 0.1px 0.5px #999999;
  text-decoration: underline;
}

.edit_comment_toggle {
  cursor: pointer;
  font-weight: bold;
  position: absolute;
  right: 2em;
}
.edit_history_modal {
  overflow: scroll;
  max-height: 500px;
  min-height: 300px;
}
.edit_comment_history_card {
    border-radius: 1em;
    border: 1px solid $primary-color;
    padding:  1em 0 1em 0;
    margin: 1em 0;
    background: #f3f3f3;

  .edited_comment_card__main__body{
    font-size: 1em;
    color: white;
    padding-left: 1em;

    p{
      width: 130px;
      background: $primary-color;
      text-align: center;
      margin-left: -2em;
      border-radius: 0.2em;
    }
  }

  span {
    font-size: 1.3em;
    display: block;
    color: #414141;
    padding: 0 1em 0 1em;
  }
}
.edit__history--loading{
  text-align: center;
  
  > img {
    width: 50%;
  }
}
.no__edit__history {
  text-align: center;
  padding: 4em 0;
}
.no__edit__history > span {
  color: #888888;
  font-size: 2em;
}
.edit_comment_toggle:hover {
  text-shadow: 0.5px 0.1px 0.5px #999999;
  text-decoration: underline;
}

@media only screen and (max-width: 1000px) {
  /* For mobile devices: */
  .comment_card {
    grid-template-columns: 20% 80%;
  }
}

@media only screen and (max-width: 768px) {
  /* For mobile devices: */
  .comment {
    width: 80%;
    &_box {
      width: 100%;
      margin-bottom: 5%;

      &_post {
        padding: 10px 0;
        float: right;
        color: #2fc1ff;
        outline-color: #cccccc;
      }

      &_post:hover {
        cursor: pointer;
      }

      &__form {
        width: 100%;

        &__submit {
          width: 30%;
        }

        &__cancel {
          width: 30%;
        }
      }
    }
  }
}

@media only screen and (max-width: 320px) {
  /* For mobile devices: */
  .comment_card {
    grid-template-columns: 25% 75%;
  }

  .edit_comment_history_card{
    text-align: center;
  }
}
