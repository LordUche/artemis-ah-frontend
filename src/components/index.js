import SideNav from './SideNav';
import TopNav from './TopNav';
import Welcome from './Welcome';

export { SideNav, TopNav, Welcome };
