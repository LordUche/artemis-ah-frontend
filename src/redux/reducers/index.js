import { combineReducers } from 'redux';
import authReducer from './authReducer';
import userReducer from './userReducer';
import profileReducer from './profileReducer';
import tagReducer from './tagReducer';
import articleReducer from './articleReducer';
import notificationsReducer from './notificationsReducer';
import commentsReducer from './commentReducer';

const rootReducer = combineReducers({
  auth: authReducer,
  user: userReducer,
  profile: profileReducer,
  tags: tagReducer,
  article: articleReducer,
  notifications: notificationsReducer,
  comments: commentsReducer,
});

export default rootReducer;
