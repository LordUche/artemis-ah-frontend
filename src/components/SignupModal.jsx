import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { func, bool } from 'prop-types';
import signUpUser from '../redux/actions/signUp';
import InputField from './InputField';
import Button from './Button';
import or from '../assets/img/Frame.png';
import Modal from './Modal';

// Images
import faceBookLogo from '../assets/img/facebooklogo.svg';
import twitterLogo from '../assets/img/twitterlogo.svg';
import googleLogo from '../assets/img/googlelogo.svg';

/**
 * @description Sign up a user
 */
export class SignUp extends Component {
  /**
   * @description constructor
   * @param {*} props
   */
  constructor(props) {
    super(props);
    this.onInputChange = this.onInputChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  state = {
    firstname: '',
    lastname: '',
    username: '',
    email: '',
    password: '',
    confirmPassword: '',
    formErrors: {
      firstname: '',
      lastname: '',
      username: '',
      email: '',
      password: '',
      confirmPassword: ''
    },
    firstnameValid: false,
    lastnameValid: false,
    usernameValid: false,
    emailValid: false,
    passwordValid: false,
    confirmPasswordValid: false,
    formValid: false,
  };

  /**
   * @description onChange method
   * @param {*} event
   * @returns {object} dunno
   */
  onInputChange(event) {
    const { target } = event;
    const { value } = target;
    const { name } = target;
    // this.setState({term: event.target.value});
    this.setState({ [name]: value }, () => { this.validateField(name, value); });
    // console.log(this.state);
  }

  /**
   * @description onSubmit method
   * @param {*} event
   * @returns {object} dunno
   */
  onSubmit(event) {
    event.preventDefault();
    const { signUp, toggleSignupModal, toggleVerify } = this.props;
    signUp(this.state, toggleSignupModal, toggleVerify);
  }

  /**
   * @description onClick method
   * @param {*} event
   * @returns {object} dunno
   */
  onClick(event) {
    event.preventDefault(this);
  }

  /**
   * @description validate input
   * @param {*} fieldName
   * @param {*} value
   * @returns {object} state
   */
  validateField(fieldName, value) {
    const {
      firstnameValid,
      lastnameValid,
      usernameValid,
      formErrors,
      emailValid,
      passwordValid,
      confirmPasswordValid,
      password
    } = this.state;
    const fieldValidationErrors = formErrors;
    let isEmailValid = emailValid;
    let isPasswordValid = passwordValid;
    let isConfirmPassword = confirmPasswordValid;
    let isFirstnameValid = firstnameValid;
    let isLastnameValid = lastnameValid;
    let isUsernameValid = usernameValid;

    switch (fieldName) {
      case 'firstname':
        isFirstnameValid = value.length >= 2 && value.match((/^[A-Za-z]+$/));
        fieldValidationErrors.firstname = isFirstnameValid ? '' : 'Firstname is invalid';
        break;
      case 'lastname':
        isLastnameValid = value.length >= 2 && value.match((/^[A-Za-z]+$/));
        fieldValidationErrors.lastname = isLastnameValid ? '' : 'Lastname is invalid';
        break;
      case 'username':
        isUsernameValid = value.length >= 2 && value.match((/^[A-Za-z]+$/));
        fieldValidationErrors.username = isUsernameValid ? '' : 'Username is invalid';
        break;
      case 'email':
        isEmailValid = value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);
        fieldValidationErrors.email = isEmailValid ? '' : 'Email is invalid';
        break;
      case 'password':
        isPasswordValid = value.length >= 8;
        fieldValidationErrors.password = isPasswordValid ? '' : 'Password must be 8 characters long';
        break;
      case 'confirmPassword':
        isConfirmPassword = value === password;
        formErrors.confirmPassword = isConfirmPassword ? '' : 'Password does not match';
        break;
      default:
        break;
    }
    this.setState({
      formErrors: fieldValidationErrors,
      firstnameValid: isFirstnameValid,
      lastnameValid: isLastnameValid,
      usernameValid: isUsernameValid,
      emailValid: isEmailValid,
      passwordValid: isPasswordValid,
      confirmPasswordValid: isConfirmPassword,
    }, this.validateForm);
  }

  /**
   * @returns {HTMLElement} login
   */
  showLogin() {
    this.toggleSignupModal();
  }

  /**
   * @returns {object} state
   */
  validateForm() {
    const {
      emailValid, firstnameValid, lastnameValid, usernameValid, passwordValid, confirmPasswordValid
    } = this.state;
    if (
      firstnameValid
      && lastnameValid
      && usernameValid
      && emailValid
      && passwordValid
      && confirmPasswordValid
    ) {
      this.setState({ formValid: true });
    }
  }

  /**
   * @returns {HTMLElement} home
   */
  // redirect() {
  //   const { history } = this.props;
  //   history.push('/');
  // }

  /**
   * @description sign up form
   * @returns {object} sign up object
   */
  render() {
    const { formErrors, formValid } = this.state;
    const { showSignup, toggleSignupModal, toggleLoginModal } = this.props;
    return (
      <Modal
        customClass="ah_signup"
        modalHeader="SIGN UP"
        showModal={showSignup}
        onClose={toggleSignupModal}
      >
        <div className="ah_signup__body">
          <p className="login_body_required_text">*(Required fields)</p>
          <form className="ah_signup__form" onSubmit={this.onSubmit}>
            <div className="ah_signup__form__input">
              <InputField
                customClass="ah_signup__form__input--signup firstname"
                placeHolder="Enter Firstname"
                onChange={this.onInputChange}
                inputType="text"
                inputName="firstname"
                required
                showRequiredAsterisk
              />
              <span className="error">{formErrors.firstname}</span>
            </div>
            <div className="ah_signup__form__input">
              <InputField
                customClass="ah_signup__form__input--signup"
                placeHolder="Enter Lastname"
                onChange={this.onInputChange}
                inputType="text"
                inputName="lastname"
                required
                showRequiredAsterisk
              />
              <span className="error">{formErrors.lastname}</span>
            </div>
            <div className="ah_signup__form__input">
              <InputField
                customClass="ah_signup__form__input--signup"
                placeHolder="Enter Username"
                onChange={this.onInputChange}
                inputType="text"
                inputName="username"
                required
                showRequiredAsterisk
              />
              <span className="error">{formErrors.username}</span>
            </div>
            <div className="ah_signup__form__input">
              <InputField
                customClass="ah_signup__form__input--signup"
                placeHolder="Enter Email"
                onChange={this.onInputChange}
                inputType="text"
                inputName="email"
                required
                showRequiredAsterisk
              />
              <span className="error">{formErrors.email}</span>
            </div>
            <div className="ah_signup__form__input">
              <InputField
                customClass="ah_signup__form__input--signup"
                placeHolder="Enter Password"
                onChange={this.onInputChange}
                inputType="password"
                inputName="password"
                required
                showRequiredAsterisk
              />
              <span className="error">{formErrors.password}</span>
            </div>
            <div className="ah_signup__form__input">
              <InputField
                customClass="ah_signup__form__input--signup"
                placeHolder="Confirm Password"
                onChange={this.onInputChange}
                inputType="password"
                inputName="confirmPassword"
                required
                showRequiredAsterisk
              />
              <span className="error">{formErrors.confirmPassword}</span>
            </div>
            <div className="ah_signup__form__wrapper">
              <Button
                customClass="ah_signup__form__button"
                btnText="Sign Up"
                // onClick={this.onSubmit}
                btnType="submit"
                isDisabled={!formValid}
              />
            </div>
          </form>
          <div>
            <p className="ah_signup__has-account">
              Already have an account?
              <button type="button" onClick={toggleLoginModal} className="ah_signup__has-account--link hasAccount"> Login Here</button>
            </p>
          </div>
          <div>
            <img src={or} alt="" className="ah_signup__or-tag" />
          </div>
          <div className="ah_signup__social">
            <Button
              customClass="ah_signup__social__facebook"
              imgSrc={faceBookLogo}
              btnText="Sign in with Facebook"
              onClick={this.onClick}
            />
            <Button
              customClass="ah_signup__social__twitter"
              imgSrc={twitterLogo}
              btnText="Sign in with Twitter"
              onClick={this.onClick}
            />
            <Button
              customClass="ah_signup__social__google"
              imgSrc={googleLogo}
              btnText="Sign in with Google+"
              onClick={this.onClick}
            />
          </div>
          <div className="ah_signup__social--small">
            <Button
              customClass="ah_signup__social__facebook"
              imgSrc={faceBookLogo}
              btnText=""
              onClick={this.onClick}
            />
            <Button
              customClass="ah_signup__social__twitter"
              imgSrc={twitterLogo}
              imgCustomClass="ah_signup__social__img"
              btnText=""
              onClick={this.onClick}
            />
            <Button
              customClass="ah_signup__social__google"
              imgSrc={googleLogo}
              btnText=""
              onClick={this.onClick}
            />
          </div>
        </div>
      </Modal>
    );
  }
}

SignUp.propTypes = {
  showSignup: bool.isRequired,
  toggleSignupModal: func.isRequired,
  toggleLoginModal: func.isRequired,
  toggleVerify: func.isRequired,
  signUp: func.isRequired,
};


/**
 * @description pass state to props
 * @param {*} state
 * @returns {object} state
 */
export function mapStateToProps(state) {
  return { state };
}

/**
 * @param {object} dispatch
 * @returns {promise} user
 */
export function mapDispatchToProps(dispatch) {
  return bindActionCreators({ signUp: signUpUser }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);
