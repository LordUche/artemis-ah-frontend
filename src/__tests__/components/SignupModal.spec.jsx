import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import ReduxPromise from 'redux-promise';
import { createStore, applyMiddleware } from 'redux';
import { SignUp } from '../../components/SignupModal';
import reducers from '../../redux/reducers';

const store = createStore(reducers, applyMiddleware(ReduxPromise));
const mockFunction = jest.fn();

describe('SignUp Modal Component', () => {
  it('should have match the given snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <SignUp
          showSignup
          toggleModal={mockFunction}
          toggleVerify={mockFunction}
          signUp={mockFunction}
          onClose={mockFunction}
        />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('shalow mount signup modal', () => {
    let signupModal;
    beforeEach(() => {
      signupModal = mount(
        <SignUp
          showModal={mockFunction}
          onClose={mockFunction}
          toggleModal={mockFunction}
          toggleVerify={mockFunction}
          showSignup
          signUp={mockFunction}
        />
      );
    });

    it('Should have form element', () => {
      expect(signupModal.find('form').exists()).toBe(true);
    });
    it('Should find input field parent div', () => {
      expect(signupModal.find('.ah_signup__form__input').exists()).toBe(true);
    });
  });
});

describe('input fields', () => {
  let signUpModal;
  beforeEach(() => {
    signUpModal = mount(
      <SignUp
        showModal={mockFunction}
        onClose={mockFunction}
        toggleModal={mockFunction}
        toggleVerify={mockFunction}
        showSignup
        signUp={mockFunction}
      />
    );
  });
  it('Should change firstname state on update', () => {
    const event = { target: { name: 'firstname', value: 'mitch' } };
    const firsnameInput = signUpModal.find('.ah_signup__form__input--signup').first();
    expect(firsnameInput.exists()).toEqual(true);
    firsnameInput.simulate('change', event);
    expect(signUpModal.state().firstname).toEqual('mitch');
  });

  it('Should change lastname state on update', () => {
    const event = { target: { name: 'lastname', value: 'mitch' } };
    const lastnameInput = signUpModal.find('.ah_signup__form__input--signup').at(1);
    expect(lastnameInput.exists()).toEqual(true);
    lastnameInput.simulate('change', event);
    // console.log(signUpModal.state());
    expect(signUpModal.state().lastname).toEqual('mitch');
  });

  it('Should change username state on update', () => {
    const event = { target: { name: 'username', value: 'mitch' } };
    const UsernameInput = signUpModal.find('.ah_signup__form__input--signup').at(2);
    expect(UsernameInput.exists()).toEqual(true);
    UsernameInput.simulate('change', event);
    expect(signUpModal.state().username).toEqual('mitch');
  });

  it('Should change email state on update', () => {
    const event = { target: { name: 'email', value: 'mitch@gmail.com' } };
    const emailInput = signUpModal.find('.ah_signup__form__input--signup').at(3);
    expect(emailInput.exists()).toEqual(true);
    emailInput.simulate('change', event);
    expect(signUpModal.state().email).toEqual(event.target.value);
  });

  it('Should change password state on update', () => {
    const event = { target: { name: 'password', value: 'oooooooo' } };
    const passwordInput = signUpModal.find('.ah_signup__form__input--signup').at(4);
    expect(passwordInput.exists()).toEqual(true);
    passwordInput.simulate('change', event);
    expect(signUpModal.state().password).toEqual(event.target.value);
  });

  it('Should change password state on update', () => {
    const event = { target: { name: 'confirmPassword', value: 'oooooooo' } };
    const confirmPasswordInput = signUpModal.find('.ah_signup__form__input--signup').at(5);

    expect(confirmPasswordInput.exists()).toEqual(true);
    confirmPasswordInput.simulate('change', event);
    expect(signUpModal.state().confirmPassword).toEqual(event.target.value);
  });
});

describe('Should ahow errors when wrong values are entered', () => {
  let signUpModal;
  beforeEach(() => {
    signUpModal = mount(
      <SignUp
        showModal={mockFunction}
        onClose={mockFunction}
        toggleModal={mockFunction}
        toggleVerify={mockFunction}
        showSignup
        signUp={mockFunction}
      />
    );
  });
  it('Should show firstname error if firstname is too short', () => {
    const event = { target: { name: 'firstname', value: 'm' } };
    const firstnameInput = signUpModal.find('.ah_signup__form__input--signup').first();
    firstnameInput.simulate('change', event);
    expect(signUpModal.state().firstname).toEqual('m');
    expect(signUpModal.state().formErrors.firstname).toEqual('Firstname is invalid');
  });

  it('Should show error if lastname is too short', () => {
    const event = { target: { name: 'lastname', value: 'm' } };
    const lastnameInput = signUpModal.find('.ah_signup__form__input--signup').at(2);
    lastnameInput.simulate('change', event);
    expect(signUpModal.state().lastname).toEqual('m');
    expect(signUpModal.state().formErrors.lastname).toEqual('Lastname is invalid');
  });

  it('Should show error if username is too short', () => {
    const event = { target: { name: 'username', value: 'm' } };
    const usernameInput = signUpModal.find('.ah_signup__form__input--signup').at(3);
    usernameInput.simulate('change', event);
    expect(signUpModal.state().username).toEqual('m');
    expect(signUpModal.state().formErrors.username).toEqual('Username is invalid');
  });

  it('Should show email error for invalid email', () => {
    const event = { target: { name: 'email', value: 'mitchgmail.com' } };
    const emailInput = signUpModal.find('.ah_signup__form__input--signup').at(3);
    expect(emailInput.exists()).toEqual(true);
    emailInput.simulate('change', event);
    expect(signUpModal.state().email).toEqual(event.target.value);
    expect(signUpModal.state().formErrors.email).toEqual('Email is invalid');
  });

  it('Should return error for invalid password', () => {
    const event = { target: { name: 'password', value: 'oooooo' } };
    const passwordInput = signUpModal.find('.ah_signup__form__input--signup').at(4);
    expect(passwordInput.exists()).toEqual(true);
    passwordInput.simulate('change', event);
    expect(signUpModal.state().password).toEqual(event.target.value);
    expect(signUpModal.state().formErrors.password).toEqual('Password must be 8 characters long');
  });

  it('Should change password state on update', () => {
    const event = { target: { name: 'confirmPassword', value: 'oooooooo' } };
    const confirmPasswordInput = signUpModal.find('.ah_signup__form__input--signup').at(5);
    expect(confirmPasswordInput.exists()).toEqual(true);
    confirmPasswordInput.simulate('change', event);
    expect(signUpModal.state().confirmPassword).toEqual(event.target.value);
    expect(signUpModal.state().formErrors.confirmPassword).toEqual('Password does not match');
  });
});

describe('Submit form', () => {
  it('Send data when Form is valid', () => {
    const signUpModal = mount(
      <SignUp
        showModal={mockFunction}
        onClose={mockFunction}
        toggleModal={mockFunction}
        toggleVerify={mockFunction}
        showSignup
        signUp={mockFunction}
      />
    );
    signUpModal.setState({ formValid: true });
    const signUpbutton = signUpModal.find('.ah_signup__form__button');
    signUpbutton.simulate('click');
    expect(mockFunction).toHaveBeenCalled();
  });
});
