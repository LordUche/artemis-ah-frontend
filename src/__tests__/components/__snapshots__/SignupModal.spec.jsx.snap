// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Modal Component should have match the given snapshot 1`] = `
<div
  className="ah_modal_backdrop_wrapper "
>
  <div
    className="ah_modal_backdrop"
    onClick={[MockFunction]}
    onKeyPress={[MockFunction]}
    role="button"
    tabIndex="0"
  />
  <div
    className="ah_modal ah_signup"
  >
    <button
      className="ah_modal_close"
      onClick={[MockFunction]}
      type="button"
    >
       Ã— 
    </button>
    <header
      className="ah_modal_header"
    >
      <img
        alt="authors haven logo"
        className="ah_modal_header_img"
        src="https://res.cloudinary.com/artemisah/image/upload/v1554320883/authorshaven/AH_logo_white.png"
      />
      <h2
        className="ah_modal_header_text"
      >
        SIGN UP
      </h2>
    </header>
    <div
      className="ah_signup__body"
    >
      <p
        className="login_body_required_text"
      >
        *(Required fields)
      </p>
      <form
        className="ah_signup__form"
      >
        <div
          className="ah_signup__form__input"
        >
          <span
            className="ah-input-required-asterisk"
          >
            *
          </span>
          <input
            className="ah-input ah_signup__form__input--signup firstname"
            name="firstname"
            onChange={[Function]}
            placeholder="Enter Firstname"
            required={true}
            type="text"
          />
          <span
            className="error"
          >
            
          </span>
        </div>
        <div
          className="ah_signup__form__input"
        >
          <span
            className="ah-input-required-asterisk"
          >
            *
          </span>
          <input
            className="ah-input ah_signup__form__input--signup"
            name="lastname"
            onChange={[Function]}
            placeholder="Enter Lastname"
            required={true}
            type="text"
          />
          <span
            className="error"
          >
            
          </span>
        </div>
        <div
          className="ah_signup__form__input"
        >
          <span
            className="ah-input-required-asterisk"
          >
            *
          </span>
          <input
            className="ah-input ah_signup__form__input--signup"
            name="username"
            onChange={[Function]}
            placeholder="Enter Username"
            required={true}
            type="text"
          />
          <span
            className="error"
          >
            
          </span>
        </div>
        <div
          className="ah_signup__form__input"
        >
          <span
            className="ah-input-required-asterisk"
          >
            *
          </span>
          <input
            className="ah-input ah_signup__form__input--signup"
            name="email"
            onChange={[Function]}
            placeholder="Enter Email"
            required={true}
            type="text"
          />
          <span
            className="error"
          >
            
          </span>
        </div>
        <div
          className="ah_signup__form__input"
        >
          <span
            className="ah-input-required-asterisk"
          >
            *
          </span>
          <input
            className="ah-input ah_signup__form__input--signup"
            name="password"
            onChange={[Function]}
            placeholder="Enter Password"
            required={true}
            type="password"
          />
          <span
            className="error"
          >
            
          </span>
        </div>
        <div
          className="ah_signup__form__input"
        >
          <span
            className="ah-input-required-asterisk"
          >
            *
          </span>
          <input
            className="ah-input ah_signup__form__input--signup"
            name="confirmPassword"
            onChange={[Function]}
            placeholder="Confirm Password"
            required={true}
            type="password"
          />
          <span
            className="error"
          >
            
          </span>
        </div>
        <div
          className="ah_signup__form__wrapper"
        >
          <button
            className="ah-btn ah_signup__form__button"
            disabled={true}
            onClick={[Function]}
            type="text"
          >
            
            Sign Up
          </button>
        </div>
      </form>
      <div>
        <p
          className="ah_signup__has-account"
        >
          Already have an account?
          <a
            className="ah_signup__has-account--link"
            href="/"
          >
             Login Here
          </a>
        </p>
      </div>
      <div>
        <img
          alt=""
          className="ah_signup__or-tag"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
      <div
        className="ah_signup__social"
      >
        <button
          className="ah-btn ah_signup__social__facebook"
          disabled={false}
          onClick={[Function]}
          type="text"
        >
          <img
            alt="Sign in with Facebook Image"
            className=""
            src={
              Object {
                "process": [Function],
              }
            }
          />
          Sign in with Facebook
        </button>
        <button
          className="ah-btn ah_signup__social__twitter"
          disabled={false}
          onClick={[Function]}
          type="text"
        >
          <img
            alt="Sign in with Twitter Image"
            className=""
            src={
              Object {
                "process": [Function],
              }
            }
          />
          Sign in with Twitter
        </button>
        <button
          className="ah-btn ah_signup__social__google"
          disabled={false}
          onClick={[Function]}
          type="text"
        >
          <img
            alt="Sign in with Google+ Image"
            className=""
            src={
              Object {
                "process": [Function],
              }
            }
          />
          Sign in with Google+
        </button>
      </div>
      <div
        className="ah_signup__social--small"
      >
        <button
          className="ah-btn ah_signup__social__facebook"
          disabled={false}
          onClick={[Function]}
          type="text"
        >
          <img
            alt=" Image"
            className=""
            src={
              Object {
                "process": [Function],
              }
            }
          />
          
        </button>
        <button
          className="ah-btn ah_signup__social__twitter"
          disabled={false}
          onClick={[Function]}
          type="text"
        >
          <img
            alt=" Image"
            className="ah_signup__social__img"
            src={
              Object {
                "process": [Function],
              }
            }
          />
          
        </button>
        <button
          className="ah-btn ah_signup__social__google"
          disabled={false}
          onClick={[Function]}
          type="text"
        >
          <img
            alt=" Image"
            className=""
            src={
              Object {
                "process": [Function],
              }
            }
          />
          
        </button>
      </div>
    </div>
  </div>
</div>
`;
